pages/demo/contentful.tsx

```diff
@@ -0,0 +1,18 @@
+import Article, { ArticleProps } from '../../components/Article'
+import getArticleByTitle from '../../utilities/getArticleByTitle'
+import { GetStaticProps } from 'next'
+import React from 'react'
+
+export default function Contentful({ title, content }: ArticleProps): JSX.Element {
```

Comment:
Check the style guide. _There are no files named anything other than `index.tsx`_. The name of the page goes on the _folder_. And do we want a `demo` folder in production? I am thinking _no_. So maybe we should create a demo branch that we can merge this PR to instead of main? Then we can dispense with the `demo` folder as the whole thing is a demo:

```sh
pages/contentful/index.tsx
```

This would then be at `the-jac.co.nz/contentful/`. But does that make sense? Aren't we revealing an _implementation_ detail in our URL? What if we switch to some other service?

My guess is that this is intended to be a generic "article" page. In that case, we will want the name of the article to be in the URL. Next.js is all set up to do this. The question is, do we want to put articles at an `articles` URL? Will this site have articles (@jakebax)? Or will we just have pages that will be updated by Jake? I think for now we should assume the latter.

So the actual folder/file structure here should use a [dynamic route](https://nextjs.org/learn/basics/dynamic-routes/page-path-external-data), right?

```sh
pages/
  [slug]/
    index.tsx
  index.tsx
```

The function should, accordingly, be `Page` or `SitePage` and the `getStaticProps` should retrieve the `slug` name from the path `params`. And as these are pages, not "articles", perhaps the component, while still using the `<article>` element, should be called `PageContent` and the props type, `PageContentProps`. Thoughts?

```typescript
export default function SitePage({ content, slug, title }: PageContentProps): JSX.Element {
  return <PageContent content={content} slug={slug} title={title} />
}

export const getStaticProps: GetStaticProps = async ({ params: { slug } = {}) => {
  const props: PageContentProps = await getPageContentBySlug(slug)

  return { props }
}
```

Titles can be duplicated so we need something unique. Every page should have a `slug` in `train-case` (because it will form part of the URL). You'll need to update Contentful to include a `slug` key-value pair.

(If it's not too onerous, I prefer to alphabetize all lists. -- that includes imports, props, attributes, etc. -- unless there is another order required.)

Reply:
OK

Reply:
So there should an error handler if the slug is not present in Contentful, right?

Reply:
If the slug is not there, then the page is not there, hence it should return null, undefined, [], or whatever (not an error, just no response). Then the SitePage should note that no data was returned and return a 404. But we can set up a page (eventually) to replace the 404 page that apologizes for not finding the page, lists the site pages and/or makes suggestions based on close matches to the slug, or even allows a search... but that's down the road. For now, if the data is not found, have the SitePage output "404 Not Found".

Reply:
I know next provides a "Next" 404 out of the box but may want to do something simple for new jac city?

Reply:
Sure.

Reply:
OK